src/hotspot/share/services/management.cpp

```diff
@@ -2097,14 +2097,18 @@ jlong Management::ticks_to_ms(jlong ticks) {
 }
 
 // Gets the amount of memory allocated on the Java heap since JVM launch.
-JVM_ENTRY(jlong, jmm_GetAllThreadAllocatedMemory(JNIEnv *env))
+JVM_ENTRY(jlong, jmm_GetTotalThreadAllocatedMemory(JNIEnv *env))
     // The result may be "too small" because the threads iterator takes a
     // snapshot and new threads may be created after result is initialized.
     // There's no check for terminated threads in the loop because their
     // allocated byte counts aren't included in exited_allocated_bytes
-    // when result is initialized.
-    jlong result = ThreadService::exited_allocated_bytes();
-    for (JavaThreadIteratorWithHandle jtiwh; JavaThread *thread = jtiwh.next();) {
+    // when result is initialized. The iterator is created and initialized
+    // before the call to exited_allocated_bytes to ensure we don't miss
+    // accounting for threads that are just about to terminate.
```

Comment:
Sorry this is only accurate if a thread updates `exited_allocated_bytes` after it removes itself from the Threads-list.

Reply:
Fixed, see below.